app/containers/App/types.d.ts

```diff
@@ -1,18 +1,19 @@
 import { ActionType } from 'typesafe-actions';
 import * as actions from './actions';
 import { ApplicationRootState } from 'types';
+import { Repo } from '../RepoListItem/types';
 
 /* --- STATE --- */
 
 interface AppState {
   readonly loading: boolean;
-  readonly error: object | boolean;
+  readonly error?: object | boolean;
   readonly currentUser: string;
   readonly userData: UserData;
 }
 
 interface UserData {
-  readonly repositories: object[] | boolean; // too many fields. Won't declare them all
+  readonly repos?: Repo[]; // too many fields. Won't declare them all
```

Comment:
regarding to #38 I would avoid renaming things. Adding types etc is really nice but I wanna stick to original code as much as possible with namings and implementation.